src/ocean/text/arguments/TimeIntervalArgs.d

```diff
@@ -508,3 +578,63 @@ unittest
     Test.test!("==")(isTimeInterval("1x"), false);
     Test.test!("==")(isTimeInterval("2xm"), false);
 }
+
+/******************************************************************************
+
+    Returns the time unit to modulu, to get rounded time
+
+    Params:
+        value = a string containing an integer and the first leter of a time
+            unit. The supported units are minutes, hours and days.
+        modulu_this = the returned value
+
+    Returns:
+        true, if the string is a valid interval
+
+******************************************************************************/
+
+public bool timeUnit ( cstring value, ref long modulu_this)
+{
+    if (value == "")
+    {
+        return false;
+    }
+
+    long result;
+    long tempValue;
+
+    char unit = value[value.length - 1];
+
+    if ( !toLong(value[0..$-1], tempValue) )
+    {
+        return false;
+    }
+
+    if ( unit == 's')
+        modulu_this = ONE_SECOND;
+    if ( unit == 'm')
+        modulu_this = SECONDS_IN_MINUTE;
+    if ( unit == 'h')
+        modulu_this = SECONDS_IN_HOUR;
+    if ( unit == 'd')
+        modulu_this = SECONDS_IN_DAY;
```

Comment:
Would in general this be better as a switch with a default option returning false?